### PR TITLE
docs: New blog post about building APISIX in SLES15

### PR DESCRIPTION
**Pull request by osaizar**
Changes:
Adds a new blog post about building APISIX in SLES 15.

### CODE REVIEW

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+```shell
+git clone https://github.com/apache/apisix.git
+cd apisix
+```
+
+## Modify the utils/install-dependencies.sh script
+Currently the `utils/install-dependencies.sh` script does not support SLES 15, so we will need to modify it slightly to add support for this distro:
+
+```shell
+wget https://raw.githubusercontent.com/osaizar/apisix/master/utils/install-dependencies.sh
```
Comment (shreemaan-abhishek) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]: I think we are better off having the shell script in the blog post itself. WDYT @yzeng25?

**Review by shreemaan-abhishek — COMMENTED** [review]
I assume following these steps in a docker container would work?

**Comment by osaizar** [comment]
> I assume following these steps in a docker container would work?

Yes, I tested it using the container. It should work.

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md (hunk 1 above)
Comment (osaizar) [reply blog/en/blog/2024/04/05/build-apisix-in-sles15.md]: > I think we are better off having the shell script in the blog post itself. WDYT @yzeng25?

I agree, that way you don't need to trust that I don't modify the script in my repository or take it down entirely.
I will make a commit fixing this.

**Comment by osaizar** [comment]
I added the modified install-dependencies.sh to the blog post. 
I think the script takes too much space in the post, this is not really an issue but feel free to suggest any other way of doing this if you think it looks weird.

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+title: "How to build APISIX in SLES 15"
+authors:
+  - name: "Oier Saizar"
+    title: "Author"
+    url: "https://github.com/osaizar"
+    image_url: "https://avatars.githubusercontent.com/u/9879984"
+keywords: 
+- Apache APISIX
+- SLES
+- Suse
```
Comment (yzeng25) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]:
```suggestion
- SUSE
```

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+description: By reading this article, you will learn how to build Apache APISIX for Suse Linux Enterprise 15
```
Comment (yzeng25) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]:
```suggestion
description: By reading this article, you will learn how to build Apache APISIX for SUSE Linux Enterprise 15
```

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        arch_path="arm64/"
+    fi
+    if [[ "${1}" == "ubuntu" ]]; then
+        sudo add-apt-repository -y "deb http://openresty.org/package/${arch_path}ubuntu $(lsb_release -sc) main"
+    elif [[ "${1}" == "debian" ]]; then
+        sudo add-apt-repository -y "deb http://openresty.org/package/${arch_path}debian $(lsb_release -sc) openresty"
+    fi
+    sudo apt-get update
+
+    # install some compilation tools
```
Comment (yzeng25) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]:
```suggestion
    # Install some compilation tools
```

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+We are ready to follow the [Building APISIX from source](https://apisix.apache.org/docs/apisix/next/building-apisix/) documentation.
+
+We can launch the next commands to build and install APISIX:
+
+```shell
+make deps
+make install
+```
+
+## Installing etcd
```
Comment (yzeng25) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]:
```suggestion
## Install etcd
```

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+ETCD_VERSION='3.4.18'
+wget https://github.com/etcd-io/etcd/releases/download/v${ETCD_VERSION}/etcd-v${ETCD_VERSION}-linux-amd64.tar.gz
+tar -xvf etcd-v${ETCD_VERSION}-linux-amd64.tar.gz && \
+  cd etcd-v${ETCD_VERSION}-linux-amd64 && \
+  sudo cp -a etcd etcdctl /usr/bin/
+  cd ..
+nohup etcd >/tmp/etcd.log 2>&1 &
+```
+
+## Creating a nobody user & group
```
Comment (yzeng25) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]:
```suggestion
## Create a nobody user and group
```

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+
+Before launching APISIX we need to create a user and a group both called `nobody`.
+This is necessary for `openresty` to launch correctly.
+
+```shell
+useradd nobody -U
+```
+
+## Running APISIX
```
Comment (yzeng25) [inline blog/en/blog/2024/04/05/build-apisix-in-sles15.md]:
```suggestion
## Run APISIX
```

File: blog/en/blog/2024/04/05/build-apisix-in-sles15.md (hunk 1 above)
Comment (yzeng25) [reply blog/en/blog/2024/04/05/build-apisix-in-sles15.md]: Sure, having the script in the blog would be good.

**Review by yzeng25 — APPROVED** [review]

**Comment by yzeng25** [comment]
@osaizar  LGTM (I'm Ex-SUSE/Rancher contributing out of business hours BTW 😄  )

**Comment by osaizar** [comment]
> @osaizar LGTM (I'm Ex-SUSE/Rancher contributing out of business hours BTW 😄 )

Thank you!

I've been using SUSE at work for some years now. I want to use APISIX but it won't be possible if there is no support 😅
I'm trying to add it myself in my off hours so we can use it, I hope it helps other people too 😄

**Review by moonming — APPROVED** [review]

**Comment by yzeng25** [comment]
@moonming please use **Squash and merge** next time.

image

**Pull request merged**